scripts/parser/backend/cmakecachevariable.py

```diff
@@ -0,0 +1,39 @@
+"""
+basic idea here is just having a special variable that deals with i/o
+and validation, while this just communicates with TUI.
+"""
+from cmakecachestring import cmakecache_string as cmc_str
+
+
+class cmakecache_variable:
+   BOOL_MAPPINGS = {
+      "ON" : True,
+      "OFF": False,
+      "0" : True,
+      "1" : False,
+      "TRUE": True,
+      "FALSE": False,
+   }
+
+   def __init__(self, line: str, comment: str, idx: int) -> None:
+      self.comment: str = comment
+      self.idx: idx = idx
```

Comment:
**Misleading name**: Index does not mean anything to me. It really should be something like `row_number` or just `row`.

Reply:
I changed `idx` to `row_number`.